src/index.js

```diff
@@ -122,6 +171,79 @@ function denormalizeEntity(entityOrId, entities, schema, bag) {
 }
 
 /**
+ * Same as denormalizeEntity right above, but memoized
+ * /!\ Does not handle circular dependencies
+ */
+// TODO: Pass cache as parameter to enable differents caches for different entity sets with entity names that may collide (e.g. 2 entity sets each with a "Post" entity)
+export let cache = {}
+
+function denormalizeEntityMemoized(entityOrId, entities, schema, bag) {
+  const key = schema.getKey();
+  const { entity, id } = resolveEntityOrId(entityOrId, entities, schema)
+
+  if (!entity)
+    return null
+  
+  /* Cache */
+  if (!cache[key])
+    cache[key] = {}
+  if (!cache[key][id]) {
+    cache[key][id] = {
+      entity,
+      denormalized: entity,
+    }
+  }
+  /* Cache *****/
+
+  if(!bag.hasOwnProperty(`${key}:${id}`)) {
+    bag[`${key}:${id}`] = true
+
+    /* If cache entity is different, wipe cache */
+    if (cache[key][id].entity !== entity) {
+      cache[key][id].entity = entity
+      cache[key][id].denormalized = entity
+    }
+
+    /* Start with the cache as reference */
+    let referenceObject = cache[key][id].denormalized
+    let relationsToUpdate = {}
+
+    /* For each relation in EntitySchema */
```

Comment:
`relation` is a relation in the entity `Schema`, which are Entities, yes 👍 . I fill this object with all the relations that have been updated - in the loop below - because we want to return the original object if nothing changed !